src/main/java/com/google/sps/servlets/ReviewsServlet.java

```diff
@@ -70,7 +71,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) {
       businessEntity.setProperty(BUSINESS_RATING, aggregatedRating);
 
       // Redirect back to the product page.
-      response.sendRedirect("/product.html?businessID="+businessKey);
+      response.sendRedirect("/product.html?businessID=%d", businessID);
```

Comment:
The formatting works by substituting `businessID `for `%d` and returns a String.
https://beginnersbook.com/2017/10/java-string-format-method/
`String redirectUrl = String.format("/product.html?businessID=%d", businessID);
response.sendRedirect(redirectUrl);`

Reply:
In that case isn't it more concise to just use +businessID? Why is it necessary to use the format specifier?

Reply:
That's true, but if we have longer strings with parameters, for example `"/product.html?businessID=" + businessID + "&searchQuery=" + searchQuery + "&foo=" + foo + `...etc, I would argue that using + for concatenation becomes hard to read. on the other hand, having `String.format("/product.html?businessID=%d&searchQuery=%s&foo=%s", businessID, searchQuery, foo)` would be more readable and less prone to errors.

Reply:
actually, on reading a [stackoverflow ](https://stackoverflow.com/questions/925423/is-it-better-practice-to-use-string-format-over-string-concatenation-in-java#:~:text=format%20gives%20you%20more%20power,format%20is%20also%20shorter.)on this, I'm undecided, since format has worse performance than concat. I think we can stick to 1 standard. The Google Java style guide also doesn't specify this.

Reply:
I don't think it's necessary to have such strict style rules! In this particular case, I'm just appending one parameter to the string, so there isn't added advantage to use the format specifier. If we constrain everyone to just using format, it's an additional rule that doesn't add much to readability and I can't think of cases where having different string formatting (concat vs format) will break codes either.

Reply:
Agree with Lilian but regardless, I personally find concat more readable. If format has worse performance, I vote concat.

Reply:
I want to merge this PR soon so the changes don't keep stacking, can I get an approval pls if this isn't a major issue?

Reply:
yes i'll approve it